### ISSUE TITLE
feat: Configurable object/column name formatting options for targets

### ISSUE THREAD

**Issue by dluo-sig** [meltano/sdk#2490]
### Feature scope

Targets (data type handling, batching, SQL object generation, tests, etc.)

### Description

It would be helpful if there is a way to configure the column and object names that get mapped from the extractor.  

1. direct mapping - names as-is
2. snake_case
3. PascalCase

Nice to have: some targets also have the ability to specify more column names than normally supported.  For example, SQL Server can have column names with additional characters when qualified with square brackets, PostgreSQL/Snowflake can do the same with double quotes, etc.  This would be an additional option to use the appropriate identifiers across the board when referencing objects/fields at the target.  It would not be desirable to always force this option, as for example, Snowflake becomes case-sensitive when double quotes are provided.

**Comment by visch** [comment]
Yes, good part is there's already a function for this https://github.com/meltano/sdk/blob/63340915751b798bade84572a6d6bda2da9e2d1b/singer_sdk/sinks/sql.py#L134

I overrode this as I wanted "direct mapping" in your list to be the default not snake_case see https://github.com/MeltanoLabs/target-postgres/blob/1e59be2750961876d52b8e69cf05c5eb06cb13b4/target_postgres/sinks.py#L280-L282 

There was a longer discussion about this here https://github.com/meltano/sdk/issues/1205

I think the idea of a config option to choose between them is a good one!

**Comment by dluo-sig** [comment]
@edgarrmondragon had proposed using the [humps ](https://github.com/nficano/humps) library for this.

**Comment by dluo-sig** [comment]
Related request https://github.com/meltano/sdk/issues/2545#issuecomment-2234055006

**Comment by lumenn** [comment]
I'm currently looking for same use case - i'd like to convert all column names to lower case (source is tap-mssql, and loader is target-postgres). In my case tables have around 100 columns, so making it manually per column doesn't seem inviting :)

**Comment by dluo-sig** [comment]
I think what could happen is that with this new configuration set, PluginMapper can take it into account in `register_raw_stream_schema` and insert key pairs into `self.stream_maps_dict` with the new mapping and a `__NULL__` for the old column.  We then need to handle `__key_properties__` to take into account key columns that were renamed.

**Comment by ReubenFrankel** [comment]
> I'm currently looking for same use case - i'd like to convert all column names to lower case (source is tap-mssql, and loader is target-postgres). In my case tables have around 100 columns, so making it manually per column doesn't seem inviting :)

Looking for something like this as well - not necessarily on the target but for `stream_maps`:

```yml
stream_maps:
  '*':
    __alias__: __property_name__.lower()
```

with a new property `__alias__` construct and scoped `__property_name__` [variable](https://sdk.meltano.com/en/v0.44.3/stream_maps.html#built-in-variable-names).

Relevant to https://sdk.meltano.com/en/v0.44.3/stream_maps.html#aliasing-properties

If `__alias__` is confusing in this context, `__map__` might be better:

```yml
stream_maps:
  '*':
    __map__: '".join("_" + char.lower() if char.isupper() else char for char in __property_name__)'  # more complex example converting camelCase to snake_case
```

**Comment by ReubenFrankel** [comment]
> stream_maps:
>   '*':
>     __alias__: __property_name__.lower()

I now realise that this conflicts with stream aliasing as currently implement, so perhaps it would need to be a new dunder construct.

**Comment by ReubenFrankel** [comment]
> stream_maps:
>   '*':
>     __alias__: __property_name__.lower()

I now realise that this conflicts with stream aliasing as currently implemented, so perhaps it would need to follow a new dunder construct...
